Intermittent integration test race failures

This appears to be coming and going in master today. I saw it surface on my PR in jenkins and locally, and switched to testing off master, it appears there sometimes as well, but sometimes not. Seems relatively easy to reproduce though, I'm seeing it every couple runs or so.

```
==================                                                                                                                                                                                                            [69/1987]
WARNING: DATA RACE                                                                                                                                                                                                                     
Read at 0x00c4208829c0 by goroutine 73:                                                                                                                                                                                                
  runtime.chansend()                                                                                                                                                                                                                         /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/chan.go:128 +0x0                                                                                                                                                                       github.com/openshift/cluster-operator/test/integration.startServerAndControllers.func1()                                                                                                                                                   /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch/watch.go:135 +0x503                                                                                                             
  github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*SimpleReactor).React()                   
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fixture.go:410 +0x64                                                                                                                   github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*Fake).Invokes()                          
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fake.go:143 +0x267
  github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake.(*FakeJobs).Create()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake/fake_job.go:82 +0x28f                                                                                         
  github.com/openshift/cluster-operator/pkg/controller.(*jobControl).createJob()                                   
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:409 +0x873          
  github.com/openshift/cluster-operator/pkg/controller.(*jobControl).ControlJobs()                                 
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:235 +0x6e6
  github.com/openshift/cluster-operator/pkg/controller.(*jobSync).Sync()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobsync.go:130 +0x54f
  github.com/openshift/cluster-operator/pkg/controller.(JobSync).Sync-fm()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/accept/accept_controller.go:108 +0x5e
  github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).processNextWorkItem()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:233 +0x145
  github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).worker()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:222 +0x38
  github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).(github.com/openshift/cluster-operator/pkg/controller/machineset.worker)-fm()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203 +0x41
  github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x6f
  github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xcd
  github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x5a

Previous write at 0x00c4208829c0 by goroutine 254:
  [failed to restore the stack]

Goroutine 73 (running) created at:
  github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).Run()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203 +0x2a6
  github.com/openshift/cluster-operator/test/integration.startServerAndControllers.func4.1()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/test/integration/controller_test.go:366 +0x53
  github.com/openshift/cluster-operator/test/integration.startServerAndControllers.func8()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/test/integration/controller_test.go:404 +0x5a

Goroutine 254 (finished) created at:
  github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x6f
  github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:53 +0xd0
  github.com/openshift/cluster-operator/vendor/k8s.io/client-go/tools/cache.(*controller).Run()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/tools/cache/controller.go:122 +0x3af
  github.com/openshift/cluster-operator/vendor/k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run()
      /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:226 +0x80c
==================
E0223 09:57:00.972004   31648 runtime.go:66] Observed a panic: "send on closed channel" (send on closed channel)
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72                                                                                                                  
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65                                                                                                                  
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51                                                                                                                  /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/asm_amd64.s:509                                                                                                                                                                            /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/panic.go:491                                                                                                                                                                               /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/chan.go:173                                                                                                                                                                                
/home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/chan.go:113                                                            
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch/watch.go:135                                                                                                                          /home/dgoodwin/go/src/github.com/openshift/cluster-operator/test/integration/controller_test.go:300                
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fixture.go:410          
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fake.go:143            
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake/fake_job.go:82                                                                                                      
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:409                       
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:235                       
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobsync.go:130                          
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/accept/accept_controller.go:108
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:233
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:222
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133      
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88                
/home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/asm_amd64.s:2337                            
panic: send on closed channel [recovered]                                                                                     
        panic: send on closed channel                                                                                                                        
                                                                                                                              
goroutine 2408 [running]:                                                                           
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)                
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x16f
panic(0x231f0a0, 0x27fa9f0)                                                                                                 
        /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/panic.go:491 +0x2a2                 
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch.(*FakeWatcher).Add(...)                         
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch/watch.go:135
github.com/openshift/cluster-operator/test/integration.startServerAndControllers.func1(0x3515dc0, 0xc42046b700, 0x4, 0xc421ec0000, 0xc42012f7d8, 0xc42196cd30, 0x2588514)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/test/integration/controller_test.go:300 +0x504
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*SimpleReactor).React(0xc42196cd20, 0x3515dc0, 0xc42046b700, 0xc421bef801, 0x80, 0x16794e8, 0x436ee0, 0xc421bef678)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fixture.go:410 +0x65
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*Fake).Invokes(0xc42012f7c0, 0x3515dc0, 0xc42046b700, 0x34f86c0, 0xc421c26c00, 0x0, 0x0, 0x0, 0x0)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fake.go:143 +0x268         
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake.(*FakeJobs).Create(0xc421959740, 0xc421c26800, 0xe, 0x351c0c0, 0xc421959740)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake/fake_job.go:82 +0x290
github.com/openshift/cluster-operator/pkg/controller.(*jobControl).createJob(0xc42194c1b0, 0xc4215b4db0, 0x28, 0x35240c0, 0xc42050c6c0, 0x34f2a80, 0xc421aed3e0, 0x3522120, 0xc420666fa0, 0x0, ...)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:409 +0x874    
github.com/openshift/cluster-operator/pkg/controller.(*jobControl).ControlJobs(0xc42194c1b0, 0xc4215b4db0, 0x28, 0x35240c0, 0xc42050c6c0, 0xc421aed301, 0x34f2a80, 0xc421aed3e0, 0xc421244300, 0xc4209a00c0, ...)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:235 +0x6e7
github.com/openshift/cluster-operator/pkg/controller.(*jobSync).Sync(0xc420242380, 0xc4215b4db0, 0x28, 0x0, 0x0)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobsync.go:130 +0x550                   
github.com/openshift/cluster-operator/pkg/controller.(JobSync).Sync-fm(0xc4215b4db0, 0x28, 0xc420432dc0, 0x226e100)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/accept/accept_controller.go:108 +0x5f   
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).processNextWorkItem(0xc421242140, 0xc420463e00)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:233 +0x146
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).worker(0xc421242140)      
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:222 +0x39   
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).(github.com/openshift/cluster-operator/pkg/controller/machineset.worker)-fm()
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203 +0x42
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc421147070)       
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:235                                                                                                                                   [0/1987]
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobsync.go:130                                                                                                                                              
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/accept/accept_controller.go:108                                                                                                                             
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:233                                                                                                                     /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:222                                                                                                                     /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203                                                                                                                     /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133                                                                                                                       
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134   
/home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88                                                                                                                        /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/asm_amd64.s:2337                                                       
panic: send on closed channel [recovered]                                                                           
        panic: send on closed channel                                                                              
                                                                                                                                                                                                                                       
goroutine 2408 [running]:                                                                                          
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)       
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x16f
panic(0x231f0a0, 0x27fa9f0)                                                                               
        /home/dgoodwin/.gvm/gos/go1.9.4/src/runtime/panic.go:491 +0x2a2                                           
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch.(*FakeWatcher).Add(...)                
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/watch/watch.go:135
github.com/openshift/cluster-operator/test/integration.startServerAndControllers.func1(0x3515dc0, 0xc42046b700, 0x4, 0xc421ec0000, 0xc42012f7d8, 0xc42196cd30, 0x2588514)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/test/integration/controller_test.go:300 +0x504
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*SimpleReactor).React(0xc42196cd20, 0x3515dc0, 0xc42046b700, 0xc421bef801, 0x80, 0x16794e8, 0x436ee0, 0xc421bef678)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fixture.go:410 +0x65
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing.(*Fake).Invokes(0xc42012f7c0, 0x3515dc0, 0xc42046b700, 0x34f86c0, 0xc421c26c00, 0x0, 0x0, 0x0, 0x0)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/testing/fake.go:143 +0x268                                       
github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake.(*FakeJobs).Create(0xc421959740, 0xc421c26800, 0xe, 0x351c0c0, 0xc421959740)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/client-go/kubernetes/typed/batch/v1/fake/fake_job.go:82 +0x290
github.com/openshift/cluster-operator/pkg/controller.(*jobControl).createJob(0xc42194c1b0, 0xc4215b4db0, 0x28, 0x35240c0, 0xc42050c6c0, 0x34f2a80, 0xc421aed3e0, 0x3522120, 0xc420666fa0, 0x0, ...)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:409 +0x874                         
github.com/openshift/cluster-operator/pkg/controller.(*jobControl).ControlJobs(0xc42194c1b0, 0xc4215b4db0, 0x28, 0x35240c0, 0xc42050c6c0, 0xc421aed301, 0x34f2a80, 0xc421aed3e0, 0xc421244300, 0xc4209a00c0, ...)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobcontrol.go:235 +0x6e7
github.com/openshift/cluster-operator/pkg/controller.(*jobSync).Sync(0xc420242380, 0xc4215b4db0, 0x28, 0x0, 0x0)           
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/jobsync.go:130 +0x550             
github.com/openshift/cluster-operator/pkg/controller.(JobSync).Sync-fm(0xc4215b4db0, 0x28, 0xc420432dc0, 0x226e100)                                                      
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/accept/accept_controller.go:108 +0x5f
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).processNextWorkItem(0xc421242140, 0xc420463e00)                                                             
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:233 +0x146
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).worker(0xc421242140)                                                                       
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:222 +0x39
github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).(github.com/openshift/cluster-operator/pkg/controller/machineset.worker)-fm()              
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203 +0x42                
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc421147070)                                                                                     
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x70
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc421147070, 0x3b9aca00, 0x0, 0x1, 0xc420448000)                                                                     
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xce
github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc421147070, 0x3b9aca00, 0xc420448000)
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x5b
created by github.com/openshift/cluster-operator/pkg/controller/machineset.(*Controller).Run                       
        /home/dgoodwin/go/src/github.com/openshift/cluster-operator/pkg/controller/machineset/machineset_controller.go:203 +0x2a7
FAIL    github.com/openshift/cluster-operator/test/integration  20.598s                                                      
make: *** [Makefile:247: test-integration] Error 1                                                                               
```

--- a/test/integration/controller_test.go
+++ b/test/integration/controller_test.go
@@ -286,7 +286,7 @@ func startServerAndControllers(t *testing.T) (
 	metav1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
 	kubefake.AddToScheme(scheme)
 	objectTracker := clientgotesting.NewObjectTracker(scheme, codecs.UniversalDecoder())
-	kubeWatch := watch.NewFake()
+	kubeWatch := watch.NewRaceFreeFake()
 	// Add a reactor for sending watch events when a job is modified
 	objectReaction := clientgotesting.ObjectReaction(objectTracker)
 	fakePtr.AddReactor("*", "jobs", func(action clientgotesting.Action) (bool, runtime.Object, error) {

